### ISSUE TITLE
[ntuple] Improve normalization of platform-specific primitives and typedefs

### ISSUE THREAD

**Issue by mxxo** [root-project/root#7875]
### Explain what you would like to see improved

Normalization of fields created through `RFieldBase::Create` does not work to the same extent as directly templated fields:
```cpp
// works 
auto ulong_field = RField<unsigned long>("ulong"); 
EXPECT_STREQ("std::uint64_t", ulong_field.GetType().c_str());
// throws runtime exception
auto other_ulong = RFieldBase::Create("other ulong", "unsigned long").Unwrap();
```
```
C++ exception with description "Field other ulong has unknown type unsigned long (unchecked RResult access!)
At:
  static ROOT::Experimental::RResult<std::unique_ptr<ROOT::Experimental::Detail::RFieldBase> > ROOT::Experimental::Detail::RFieldBase::Create(const string&, const string&) [/home/max/projects/rootdev/root/tree/ntuple/v7/src/RField.cxx:202]
```
Typedef / type alias fields are also an issue:
```cpp
// works
auto str_size_field = RField<std::string::size_type>("str_size");
// throws exception
auto other_str_size = RFieldBase::Create("str_size", "std::string::size_type").Unwrap();
```
```
C++ exception with description "Field str_size has unknown type std::string::size_type (unchecked RResult access!)
At:
  static ROOT::Experimental::RResult<std::unique_ptr<ROOT::Experimental::Detail::RFieldBase> > ROOT::Experimental::Detail::RFieldBase::Create(const string&, const string&) [/home/max/projects/rootdev/root/tree/ntuple/v7/src/RField.cxx:202]
```
This has implications for user-defined classes, because `RFieldBase::Create` is used to initialize subfields:
```cpp
struct CustomStruct {
   unsigned long ul;
};
// ... 
auto class_field = RField<CustomStruct>("custom_struct");
```
```
C++ exception with description "Field ul has unknown type unsigned long (unchecked RResult access!)
At:
  static ROOT::Experimental::RResult<std::unique_ptr<ROOT::Experimental::Detail::RFieldBase> > ROOT::Experimental::Detail::RFieldBase::Create(const string&, const string&) [/home/max/projects/rootdev/root/tree/ntuple/v7/src/RField.cxx:202]
```

### Additional context
Related issue #7861

**Comment by mxxo** [comment]
@jalopezg-r00t I think this issue is also resolved by type-translation map in #9351?

**Comment by mxxo** [comment]
Ok, how about leaving it as a issue for type-translation map completeness? Looking at the map I think these could be added: 
* `long int`
* `unsigned long int`
* `short int`
* `short unsigned int`

Not sure what to do about `long long`, `long double` etc. 

Not to mention `long long long`: https://godbolt.org/z/qvv313KTW

**Comment by jblomer** [comment]
Fixed by #16039

**Comment by hahnjo** [comment]
Yes, `RIntegralTypeMap` in https://github.com/root-project/root/pull/16039 takes care of platform specific lengths of `long` etc. I'm not sure if all weird spellings of standard integer types are handled, but I think we can close this issue for now.
